proxy_test.go

```diff
@@ -768,6 +768,35 @@ func TestShimSendStdinAfterExeccmd(t *testing.T) {
 	rig.Stop()
 }
 
+// This test ensures that a shim will receive the exit code signal even if
+// the process inside the corresponding container has not been started.
+// This case occurs when a container has been created and it receives a KILL
+// or TERM signal. We don't forward the signal to hyperstart because nothing
+// runs inside the VM, but we still need to remove the shim which is seen as
+// the container process from any orchestrator.
+func TestShimKillSignalProcessNotStarted(t *testing.T) {
+	rig := newTestRig(t)
+	rig.Start()
+
+	token := rig.RegisterVM()
+	shim := rig.ServeNewShim(token)
+	session := peekIOSession(rig.proxy, token)
+
+	session.containerID = testContainerID
+
+	// Don't close session.processStarted because we want to simulate
+	// a case where the container has been created but not started.
+
+	// Send KILL signal and check the shim receives an exit code.
+	err := shim.client.Kill(syscall.SIGKILL)
```

Comment:
Could you also add tests for `SIGTERM`, `SIGWINCH` and maybe one other signal that isn't explicitly considered by this PR (`SIGINT`?)